lib/_http_outgoing.js

```diff
@@ -289,6 +310,34 @@ OutgoingMessage.prototype._buffer = function _buffer(data, encoding, callback) {
 };
 
 
+OutgoingMessage.prototype.cork = function cork() {
+  this._corked++;
+  if (!this.connection) {
+    this.once('connection', (stream) => {
+      stream.cork();
+    });
```

Comment:
Minor nit: perhaps make this a regular top level function to avoid creating the closure?